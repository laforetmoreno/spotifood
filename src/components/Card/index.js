import React from 'react';
import { string } from 'prop-types';

import Button from '../Button';

import styles from './index.module.scss';

const Card = ({ description, image, link }) => (
  <li data-testid="card" className={styles.wrapper}>
    <Button target="_blank" link={link}>
      <span className={styles.description}>{description}</span>
      <img className={styles.image} src={image} alt={description} />
    </Button>
  </li>
);

Card.propTypes = {
  description: string,
  image: string,
  link: string,
};

Card.defaultProps = {
  description: '',
  image: '',
  link: '',
};

export default Card;
