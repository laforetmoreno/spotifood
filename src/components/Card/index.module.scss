@import '../../styles';

.wrapper {
  @include flex-column();
  margin: rem(20px) 0;
  padding: rem(20px);
  background: $--color-dark-blue;
  border-radius: rem(4px);
}

.description {
  margin-bottom: rem(20px);
}

.image {
  max-width: rem(300px);
  border-radius: rem(4px);
}
