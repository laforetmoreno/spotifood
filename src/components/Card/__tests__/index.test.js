import React from 'react';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Card from '../index';

describe('<Card />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('get by id and change', async () => {
    const { getByTestId } = render(<Card />);
    const fieldNode = await waitForElement(() => getByTestId('card'));
    fireEvent.change(fieldNode);
  });
});
