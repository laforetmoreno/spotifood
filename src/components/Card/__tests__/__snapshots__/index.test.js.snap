// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<li
  className="wrapper"
  data-testid="card"
>
  <button
    className="wrapper"
    data-testid="btn"
    onClick={[Function]}
    type="button"
  >
    <span
      className="description"
    >
      
    </span>
    <img
      alt=""
      className="image"
      src=""
    />
  </button>
</li>
`;
